services/queue/src/api.js

```diff
@@ -1746,15 +1731,11 @@ builder.declare({
   const {provisionerId, workerType} = req.params;
 
   const [wType, provisioner] = await Promise.all([
-    this.WorkerType.load({
-      provisionerId,
-      workerType,
-      expires: Entity.op.greaterThan(new Date()),
-    }, true),
-    this.Provisioner.load({provisionerId}, true),
+    WorkerType.get(this.db, provisionerId, workerType),
+    Provisioner.get(this.db, provisionerId),
   ]);
 
-  if (!wType || !provisioner) {
+  if (!wType || wType.expires < new Date() || !provisioner) {
```

Comment:
If we modify the get function to also take expires as an argument then there shouldn't be a need for this additional logic in the JS. I think we are trying to move away from adding additional business logic outside the db.

Reply:
I've changed the `get_` functions, singular and plural, to accept `expires`. In one case, we're doing more complicated logic (checking quarantine and expired status), my impression was that would need either a special case db function or add significant complexity to the current ones.

Reply:
I think we can fix this up later, if necessary.  This API may not stick around for along if we start mixing queue and worker-manager information about workers.